sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/UserAgentInterceptor.java

```diff
@@ -0,0 +1,152 @@
+        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
+        if (telemetryDisabled) {
+            this.userAgent = String.format(DISABLED_TELEMETRY_USER_AGENT_FORMAT, sdkName, sdkVersion,
+                version.getVersion());
+        } else {
+            if (applicationId == null) {
+                this.userAgent = String.format(USER_AGENT_FORMAT, sdkName, sdkVersion, getPlatformInfo(),
+                    version.getVersion());
+            } else {
+                this.userAgent = String.format(APPLICATION_ID_USER_AGENT_FORMAT, applicationId, sdkName, sdkVersion,
+                    getPlatformInfo(), version.getVersion());
+            }
+        }
+    }
+
+    /**
+     * Creates a UserAgentInterceptor with the {@code sdkName} and {@code sdkVersion} in the User-Agent header value.
+     * <p>
+     * If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be included
+     * in the user agent.
+     *
+     * @param sdkName       Name of the client library.
+     * @param sdkVersion    Version of the client library.
+     * @param version       {@link ServiceVersion} of the service to be used when making requests.
+     * @param configuration Configuration store that will be checked for the AZURE_TELEMETRY_DISABLED.
+     * @throws NullPointerException if {@code configuration} is {@code null}.
+     */
+    public UserAgentInterceptor(String sdkName, String sdkVersion, Configuration configuration, ServiceVersion version) {
+        Objects.requireNonNull(configuration, "'configuration' cannot be null.");
+        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
+        if (telemetryDisabled) {
+            this.userAgent = String.format(DISABLED_TELEMETRY_USER_AGENT_FORMAT, sdkName, sdkVersion,
+                version.getVersion());
+        } else {
+            this.userAgent = String.format(USER_AGENT_FORMAT, sdkName, sdkVersion, getPlatformInfo(),
+                version.getVersion());
+        }
+    }
+
+    /**
+     * Updates the "User-Agent" header with the value supplied in the interceptor.
+     * <p>
+     * When the User-Agent header already has a value and it differs from the value used to create this interceptor the
+     * User-Agent header is updated by prepending the value in this interceptor.
+     * {@inheritDoc}
+     */
+    @NonNull
+    @Override
+    public Response intercept(@NonNull Chain chain) throws IOException {
+        Request request = chain.request();
+        String header = request.header("User-Agent");
+
+        if (header == null || header.contains(DEFAULT_USER_AGENT_HEADER)) {
+            header = userAgent;
+        } else {
+            header = userAgent + " " + header;
+        }
+
+        return chain.proceed(request
+            .newBuilder()
+            .addHeader("User-Agent", header)
```

Comment:
This should be `.header()` except for circumstances where the header is multiply-valued (rare, e.g. Cookie)